sql/core_clients_daily_v1.sql

```diff
@@ -1,23 +1,52 @@
-CREATE TEMP FUNCTION udf_mode_last(x ANY TYPE) AS ((
-  SELECT
-    val
-  FROM (
+CREATE TEMP FUNCTION
+  udf_json_mode_last(list ANY TYPE) AS ((
+    SELECT
+      ANY_VALUE(_value)
+    FROM
+      UNNEST(list) AS _value
+    WITH
+    OFFSET
+      AS _offset
+    GROUP BY
+      TO_JSON_STRING(_value)
+    ORDER BY
+      COUNT(_value) DESC,
+      MAX(_offset) DESC
+    LIMIT
+      1));
+  --
+CREATE TEMP FUNCTION
+  udf_mode_last(list ANY TYPE) AS ((
     SELECT
-      val,
-      COUNT(val) AS n,
-      MAX(offset) AS max_offset
+      _value
     FROM
-      UNNEST(x) AS val
-    WITH OFFSET AS offset
+      UNNEST(list) AS _value
+    WITH
+    OFFSET
+      AS
+    _offset
     GROUP BY
-      val
+      _value
     ORDER BY
-      n DESC,
-      max_offset DESC
-  )
-  LIMIT 1
-));
-
+      COUNT(_value) DESC,
+      MAX(_offset) DESC
+    LIMIT
+      1 ));
+  --
+CREATE TEMP FUNCTION
+  udf_require_whole_geo(country STRING,
+    city STRING,
+    geo_subdivision1 STRING,
+    geo_subdivision2 STRING) AS ( IF(country != '??',
```

Comment:
When there's an else value, I'd prefer we avoid a negation in the condition, so this would become `IF(country = '??', NULL, STRUCT...`.

Reply:
`IF(x != '??', y, z)` isn't equivalent to `IF(x = '??', z, y)` because `NULL != '??'` returns `NULL` and `IF(NULL, y, z)` returns `z`

Reply:
so I guess the question is: which do you prefer :one: `IF(country != '??', ..., NULL)` or :two: `IF(country IS NULL OR country = '??', NULL, ...)` ?

Reply:
That was not obvious to me. I'd prefer an explicit `IF(country IS NULL OR country = '??'` if possible.